ros_packages/voice_assistant/voice_assistant/audio_player.py

```diff
@@ -111,14 +114,27 @@ def __init__(self, playback_queue: Queue[PlaybackItem]):
 
 
     def counter_next(self) -> int:
-
         with self.counter_lock:
             value = self.counter
             self.counter += 1
             return value
 
 
     
+    def adjust_data_granularity(self, data: Iterable[bytes], target_bytes_per_chunk: int) -> Iterable[bytes]:
+        """returns a new iterable, whose data chunks have the specified target size (in bytes)"""
+        data_buffer: bytes = b''
+        for chunk in data:
+            data_buffer = data_buffer + chunk
+            num_iters = len(data_buffer) // target_bytes_per_chunk
+            for i in range(num_iters):
+                offset = i*target_bytes_per_chunk
+                yield data_buffer[offset:(offset + target_bytes_per_chunk)]
+            data_buffer = data_buffer[(num_iters*target_bytes_per_chunk):]
```

Comment:
Maybe add one or two comments to understand the function more quickly

Reply:
comments were added